contracts/protocol/facets/SequenatialCommitHandlerFacet.sol

```diff
@@ -0,0 +1,195 @@
+        uint256 escrowAmount;
+        {
+            // Get sequential commits for this exchange
+            SequentialCommit[] storage sequentialCommits = protocolEntities().sequentialCommits[_exchangeId];
+
+            {
+                // Calculate fees
+                uint256 protocolFeeAmount = tokenAddress == protocolAddresses().token
+                    ? protocolFees().flatBoson
+                    : (protocolFees().percentage * actualPrice) / 10000;
+
+                // Calculate royalties
+                (, uint256 royaltyAmount) = IBosonVoucher(protocolLookups().cloneAddress[offer.sellerId]).royaltyInfo(
+                    _exchangeId,
+                    actualPrice
+                );
+
+                // Verify that fees and royalties are not higher than the price.
+                require((protocolFeeAmount + royaltyAmount) <= actualPrice, FEE_AMOUNT_TOO_HIGH);
+
+                // Get price paid by current buyer
+                uint256 len = sequentialCommits.length;
+                uint256 currentPrice = len == 0 ? offer.price : sequentialCommits[len - 1].price;
+
+                // Calculate the minimal amount to be kept in the escrow
+                escrowAmount = Math.max(actualPrice, protocolFeeAmount + royaltyAmount + currentPrice) - currentPrice;
+
+                // Update sequential commit
+                sequentialCommits.push(
+                    SequentialCommit({
+                        resellerId: buyerId,
+                        price: actualPrice,
+                        protocolFeeAmount: protocolFeeAmount,
+                        royaltyAmount: royaltyAmount
+                    })
+                );
+            }
+
+            // Make sure enough get escrowed
+            if (_priceDiscovery.side == Side.Ask) {
+                if (escrowAmount > 0) {
+                    // Price discovery should send funds to the seller
+                    // Nothing in escrow, need to pull everything from seller
+                    if (tokenAddress == address(0)) {
+                        // If exchange is native currency, seller cannot directly approve protocol to transfer funds
+                        // They need to approve wrapper contract, so protocol can pull funds from wrapper
+                        // But since protocol otherwise normally operates with native currency, needs to unwrap it (i.e. withdraw)
+                        FundsLib.transferFundsToProtocol(address(weth), seller, escrowAmount);
+                        weth.withdraw(escrowAmount);
+                    } else {
+                        FundsLib.transferFundsToProtocol(tokenAddress, seller, escrowAmount);
+                    }
+                }
+            } else {
+                // when bid side, we have full proceeds in escrow. Keep minimal in, return the difference
+                if (tokenAddress == address(0)) {
+                    tokenAddress = address(weth);
```

Comment:
After 145, the address is put directly into `transferFundsToProtocol`, and we don't need it later anymore so there is no need to update the `tokenAddress` value.
Here in line 158, I need to set it since it's used again in 163.

If I used the same pattern in lines 145-153 as I did in lines 157-163, I could get rid of 1 `FundsLib.transferFundsToProtocol`, but then I would need another if to see if protocol needs to call `weth.withdraw`. But since withdraw must be called after `transferFundsToProtocol`, `tokenAddress` would not be 0 anymore, I would need an additional flag to distinguish both cases.